rakan/cRakan/tst/test_runner.cc

```diff
@@ -8,20 +10,564 @@
 
 namespace rakan {
 
-TEST(Test_Runner, TestBFS) {
-  Graph *g = new Graph;
-  Runner r;
-  unordered_set<Node *> unused;
-
-  // 4 nodes, all in district 1
-  Node *n1 = new Node(1, 1, 0, 0);
-  Node *n2 = new Node(2, 1, 0, 0);
-  Node *n3 = new Node(3, 1, 0, 0);
-  Node *n4 = new Node(4, 1, 0, 0);
-  unused.insert(n1);
-  unused.insert(n2);
-  unused.insert(n3);
-  unused.insert(n4);
+Runner* Generate4x4Map() {
+  Runner *r = new Runner(16, 4);
+  for (uint32_t i = 0; i < 17; i++) {
+    r->add_node(i, 0, 0, 0);
+  }
+  r->add_edge(0, 1);
+  r->add_edge(1, 2);
+  r->add_edge(2, 3);
+  r->add_edge(0, 4);
+  r->add_edge(1, 5);
+  r->add_edge(2, 6);
+  r->add_edge(3, 7);
+  r->add_edge(4, 5);
+  r->add_edge(5, 6);
+  r->add_edge(6, 7);
+  r->add_edge(4, 8);
+  r->add_edge(5, 9);
+  r->add_edge(6, 10);
+  r->add_edge(7, 11);
+  r->add_edge(8, 9);
+  r->add_edge(9, 10);
+  r->add_edge(10, 11);
+  r->add_edge(8, 12);
+  r->add_edge(9, 13);
+  r->add_edge(10, 14);
+  r->add_edge(11, 15);
+  r->add_edge(12, 13);
+  r->add_edge(13, 14);
+  r->add_edge(14, 15);
+
+  vector<uint32_t> districts(16);
+
+  // First quadrant in district 0
+  districts[0] = 0;
+  districts[1] = 0;
+  districts[4] = 0;
+  districts[5] = 0;
+
+  // Second quadrant in district 1
+  districts[2] = 1;
+  districts[3] = 1;
+  districts[6] = 1;
+  districts[7] = 1;
+
+  // Third quadrant in district 2
+  districts[8] = 2;
+  districts[9] = 2;
+  districts[12] = 2;
+  districts[13] = 2;
+
+  // Fourth quadrant in district 3
+  districts[10] = 3;
+  districts[11] = 3;
+  districts[14] = 3;
+  districts[15] = 3;
+
+  r->set_districts(districts);
+  return r;
+}
+
+TEST(Test_Runner, TestRedistrict) {
```

Comment:
You'd want to use a test fixture and break this test into the # of assert statements. This makes catching bugs easier (if a test fails, it's immediately obvious which assert statement triggered the fail). But it's fine for now, just a heads up in the future.